packages/components/src/components/assets/AssetShareIndicator/AssetShareIndicator.tsx

```diff
@@ -101,15 +195,31 @@ export const AssetShareIndicator = ({
     size = 32,
     percentageShare,
     index,
+    coingeckoId = '',
+    contractAddress,
+    quality = 'medium',
     ...rest
-}: AssetShareIndicatorProps) => (
-    <Container className={className}>
-        <CoinLogo symbol={symbol} size={size} {...rest} />
-        <ProgressCircle
-            symbol={symbol}
-            size={size}
-            percentageShare={percentageShare}
-            index={index}
-        />
-    </Container>
-);
+}: AssetShareIndicatorProps) => {
+    const iconUrl = useAssetUrl(coingeckoId, contractAddress);
+    const coinLogoSize = !symbol ? QUALITY_SIZE[quality] : size;
```

Comment:
Inverted logic (`!symbol`) is a bit confusion. Its easier to read and think about it when I don't need to negate stuff in my mind.

Reply:
-> 

```ts
const coinLogoSize = symbol ? size : QUALITY_SIZE[quality];
```